src/sas/qtgui/Utilities/MuMag/MuMagLib.py

```diff
@@ -0,0 +1,558 @@
+import numpy as np
+import matplotlib.pylab as pl
+import os
+import os.path
+from datetime import datetime
+
+import scipy.optimize
+
+from PySide6 import QtWidgets
+from PySide6.QtWidgets import QFileDialog
+
+from sas.qtgui.Utilities.MuMag.datastructures import ExperimentalData, LoadFailure, FitFailure, ExperimentGeometry, \
+    FitParameters, FitResults, LeastSquaresOutputParallel, LeastSquaresOutputPerpendicular, SweepOutput
+
+from sasdata.dataloader.loader import Loader
+
+
+class MuMagLib:
+    """ Library for methods supporting MuMag"""
+
+    mu_0 = 4 * np.pi * 1e-7
+
+    @staticmethod
+    def directory_popup():
+        directory = QFileDialog.getExistingDirectory()
+
+        if directory.strip() == "":
+            return None
+        else:
+            return directory
+
+
+    @staticmethod
+    def import_data(directory):
+        """Import experimental data and get information from filenames"""
+
+        try:
+            # Load the data
+            loader = Loader()
+            input_names = [name for name in os.listdir(directory) if name.lower().endswith(".csv")]
+            input_paths = [os.path.join(directory, filename) for filename in input_names]
+
+            input_data = loader.load(input_paths)
+
+            data = []
+            for filename, data1d in zip(input_names, input_data):
+
+                # Extract the metadata from the filename
+                filename_parts = filename.split(".")
+                filename = ".".join(filename_parts[:-1])
+
+                parts = filename.split("_")
+
+                applied_field = float(parts[1])  # mT
+                saturation_magnetisation = float(parts[2])  # mT
+                demagnetising_field = float(parts[3])  # mT
+
+                # Create input data object
+                data.append(ExperimentalData(
+                    scattering_curve=data1d,
+                    applied_field=applied_field,
+                    saturation_magnetisation=saturation_magnetisation,
+                    demagnetising_field=demagnetising_field))
+
+            return sorted(data, key=lambda x: x.applied_field)
+
+        except Exception as e:
+            raise LoadFailure(f"Failed to load from {directory}: " + repr(e))
```

Comment:
Any error thrown from any file, regardless the number of successful data loads, will throw away all loaded data. If the try/catch was around the data loading loop, the filename and error could be stored until the data loading is complete and then log the issue.